src/Stratis.Bitcoin.Features.PoA/Voting/PollsCollection.cs

```diff
@@ -37,6 +37,8 @@ System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator()
 
         public void Add(Poll poll)
         {
+            Guard.Assert(!this.polls.Contains(poll));
```

Comment:
Im wondering whether we should perhaps LogWarning the poll details and rather just return? So that we know about a poll that was attempting to be re-added?

Reply:
Done